[Doc] Add Architecture decisions (#6356)
marmelab/react-admin: docs/Readme.md

See the [Data Providers documentation](./DataProviders.md) for details.

## Architecture: Batteries Included But Removable

React-admin is designed as a library of loosely coupled React components built on top of [material-ui](https://material-ui.com/), in addition to React hooks allowing to reuse the logic with a custom UI. 

> It is very easy to replace one part of react-admin with your own, e.g. to use a custom Datagrid, GraphQL instead of REST, or Bootstrap instead of Material Design.

Reviewer: ```suggestion
You may replace one part of react-admin with your own, e.g. to use a custom Datagrid, GraphQL instead of REST, or Bootstrap instead of Material Design.
```

Let's avoid simple, easy, etc.
